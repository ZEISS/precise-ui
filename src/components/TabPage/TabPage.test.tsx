import * as React from 'react';
import * as enzyme from 'enzyme';
import { TabPage } from './';

describe('<TabPage />', () => {
  it('should render an empty <TabPage> component', () => {
    const wrapper = enzyme.shallow(<TabPage header="Foo" />);
    expect(wrapper.text()).toBe('');
  });

  it('should render an some text as children', () => {
    const wrapper = enzyme.shallow(<TabPage header="Foo">Bar</TabPage>);
    expect(wrapper.text()).toBe('Bar');
  });

  it('should render a header as a component', () => {
    const wrapper = enzyme.mount(<TabPage header={<div>Header</div>}>Content of the tab</TabPage>);
    expect(wrapper).toMatchSnapshot();
  });
});
