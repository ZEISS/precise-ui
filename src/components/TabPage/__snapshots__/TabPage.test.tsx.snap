// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPage /> should render a header as a component 1`] = `
<TabPage
  header={
    <div>
      Header
    </div>
  }
>
  <div>
    Content of the tab
  </div>
</TabPage>
`;
